spec/mongoid/reloadable_spec.rb

```diff
@@ -353,5 +353,82 @@
         expect(reloaded.readonly?).to be(false)
       end
     end
+
+    context 'on sharded cluster' do
+      require_topology :sharded
+
+      before(:all) do
+        CONFIG[:clients][:other] = CONFIG[:clients][:default].dup
+        CONFIG[:clients][:other][:database] = 'other'
+        Mongoid::Clients.clients.values.each(&:close)
+        Mongoid::Config.send(:clients=, CONFIG[:clients])
+        Mongoid::Clients.with_name(:other).subscribe(Mongo::Monitoring::COMMAND, EventSubscriber.new)
+      end
+
+      let(:subscriber) do
+        client = Mongoid::Clients.with_name(:other)
+        monitoring = if client.respond_to?(:monitoring, true)
+          client.send(:monitoring)
+        else
+          # driver 2.5
+          client.instance_variable_get('@monitoring')
+        end
+        subscriber = monitoring.subscribers['Command'].find do |s|
+          s.is_a?(EventSubscriber)
+        end
+      end
+
+      let(:find_events) do
+        find_events = subscriber.started_events.select { |event| event.command_name.to_s == 'find' }
+      end
+
+      before do
+        subscriber.clear_events!
+      end
+
+      context 'without embedded document' do
+        let(:profile) do
+          Profile.with(client: :other) do |klass|
+            klass.create
+          end
+        end
+
+        before do
+          Profile.with(client: :other) do |_|
```

Comment:
I believe you can simply omit the block argument if you don't need it in the block.